Add optional additional entities to authz request

sliiser opened the pull request:
Added an option to pass in entities to authorization requests that would not overwrite the configuration already stored in memory, but modify it by adding new entities to it.

Should be used if your use case has a large amount of entities that is infeasible to store in memory and you have easy access to all relevant entities when making the authorization request.

Fixes #22

This is my first Rust PR. Open and welcoming to feedback.

Would have added tests if the architecture was set up already. Sadly, for integration tests it is not and unit tests are meaningless here.

omer9564 reviewed (commented):
It looks good in general.
I understand that you didn't implement tests because this happens in the API layer and we currently only have tests for the service layer.
I do think you can still create tests here, 
you can extract the new behavior to a function and create unit test for that function.

besides that, we currently don't have CI to run the test so we run them manually, please attach a screenshot of the successfully running tests after your changes :)

sliiser commented:
Extracted the new behaviour and the existing entities override to a function and added a test.

Attached is console output of a test run. Plenty of warnings unrelated to these changes, but all tests pass.

[test_run_output.txt](https://github.com/permitio/cedar-agent/files/13817346/test_run_output.txt)

omer9564 reviewed (approved):
LGTM, will merge it this week :)
